### DISCUSSION TITLE
MM2 do not replicate message from new broker

### DISCUSSION THREAD

**Discussion by SimonCardaire** [strimzi/strimzi-kafka-operator#7965] [Q&A]
Hi, I think I found a bug on mm2 for strimzi 0.25

MM2 do not mirror message when leader broker of the partition is a newly added broker in the source cluster

**To Reproduce**
1. Create a basic kafka cluster called source and a basic kafka cluster called target with Strimzi 0.25 (3 broker per cluster for example)
2. Create a MM2 ressource between target and source cluster
3. Expand source cluster to add one more broker 
5. Create a topic on source cluster compliant with your MM2 replication pattern
6. Publish data in the topic

**Expected behavior**
I which all data are replicated from source cluster to target cluster. But when the new added broker is leader of a partition, the data in this partition are not replicated.

For example : 

If I have 4 partition on my 4 broker with a replication factor of 3
Topic: foo   Partition: 0    Leader: 0       Replicas: 1,2 
Topic: foo   Partition: 1    Leader: 1       Replicas: 0,2 
Topic: foo   Partition: 2    Leader: 2       Replicas: 1,3 
Topic: foo   Partition: 3    Leader: 3       Replicas: 1,2 

The broker 0 1 and 2 are the initial brokers of my cluster and the broker 3 is the newly added broker. Broker 3 is added after the creation of the MM2.
All data in the Partition 3 are not replicated to the target cluster.

Is it a known issue?

Regards.

**Comment by scholzj** [comment]
This sounds more like Kafka bug than a Strimzi bug. But without the logs it might be hard to track it. 

But TBH Strimzi 0.25 (which supported Kafka 2.7 and 2.8 IIRC) is quite old and MM2 was still quite new. So a lot improved since. I would suggest to first try if you have the same issue when upgrading.
